### ISSUE TITLE
Closes after ~5 mins (Mazda Binary)

### ISSUE THREAD

**Issue by Naomarik** [JonesChi/CastScreen#1]
Seems recently you committed [Fix socket closed by wrong no-data state ](https://github.com/JonesChi/CastScreen/commit/0957877cb677161b66f0c8ac72586c2e35e3257c). 

I'm wondering if this solves the issue. The screen cast works brilliantly for about mins or so and just stops without warning. Rebooting the infotainment system is only thing that works for me. Does your latest commit solve this issue? 

If it does, would you mind compiling another binary I can drop into my Mazda? I don't know how to compile for arm.

Thanks a ton for your work on this!

**Comment by JonesChi** [comment]
Yes, the issue is fixed by the commit, and I will release a new version when I'm ready.

**Comment by JonesChi** [comment]
Hi,
Here is the new version of castscreen_receiver for mazda infotainment system.
In this version, I use adb reverse instead of forward, please follow the steps to test the new version.
[castscreen_mazda_20160308.tgz](https://mega.nz/#!iBAmmJyJ!Kh1ApiAJAwgSt9lu71IEB2oBf-t0kdTrRJ7rkqlIQww)

Steps to try CastScreen on your infotainment system.
1. Download and decompress the file "castscreen_mazda_20160308.tgz.tgz"
2. Install the CastScreen(castscreen-1.0.apk) Android APP on your Android device and enable debug mode on your device
3. Push the "adb", "cs_receiver_arm", "wait_adb_arm.sh" into "/jci/scripts" on your Mazda infotainment system
4. Change mode of these files "# chmod 755 adb wait_adb_arm.sh cs_receiver_arm"
5. Add "watch /jci/scripts/wait_adb_arm.sh &" into /jci/scripts/stage_wifi.sh  (for auto start the castscreen reveiver)
6. Execute "# ./wait_adb_arm.sh" or reboot
7. Connect your Android device with USB cable
8. On Android, launch CastScreen, change setting to H264, 800x600@160, and 1 Mbps, then input **127.0.0.1** and press input receiver, then tap Start on right-up corner
9. Here you go

**Comment by Naomarik** [comment]
Thanks! Will give you feedback once I try it out.

**Comment by Naomarik** [comment]
It's been working well so far, I did have one instance where the CastScreen app crashed and left my Mazda infotainment stuck with a frozen casting screen. I tried everything I could to get it out of it unsuccessfully. 

Disconnecting/reconnecting the phone would cause CastScreen to crash again sometimes.

This is a bit unhelpful since I don't know how to reproduce nor did I have any the wifi on to SSH in. I'm wondering if the cs_receiver_arm binary has some way of killing itself if it does not get any data from the phone after a certain period; I suspect this file got caught up in a loop.

This only happened once in about 5 drives using it.
